.metadata-tools/metadataanalyzercreator.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+	flag.Parse()
+	root := flag.Arg(0)
+
+	fset := token.NewFileSet()
+	pkgs := make(map[string]string)
+
+	err := filepath.WalkDir(root, func(path string, file fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if file.IsDir() {
+			if file.Name() == "vendor" || file.Name() == "tests" || file.Name() == "internal" {
+				return fs.SkipDir
+			}
+			return nil
+		}
+
+		if filepath.Ext(path) != ".go" {
+			return nil
+		}
+
+		componentType := ""
+		packageName := ""
+		skip := true
+		dir := filepath.Dir(path)
+		for dir != "." && !strings.HasSuffix(dir, "components-contrib") {
+			if !skip {
+				packageName = filepath.Base(dir) + "/" + packageName
+			} else {
+				packageName = filepath.Base(dir)
+			}
+			skip = false
+			dir = filepath.Dir(dir)
+
+			switch filepath.Base(dir) {
+			case "bindings", "pubsub", "state", "secretstores", "middleware", "workflows", "lock", "configuration", "crypto", "nameresolution":
```

Comment:
Can you use `.build-tools/component-folders.json` instead?

Reply:
I will take a look. Wasn't aware of that file